processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/Analyser.java

```diff
@@ -0,0 +1,144 @@
+        fileBuilder.setFilename(gtfsFileAnnotation.value());
+        fileBuilder.setSingleRow(gtfsFileAnnotation.singleRow());
+        fileBuilder.interfacesBuilder().add(type.asType());
+        fileBuilder.setClassName(entityImplementationSimpleName(type.getSimpleName().toString()));
+        fileBuilder.setRequired(type.getAnnotation(Required.class) != null);
+        for (ExecutableElement method : methodsIn(type.getEnclosedElements())) {
+            GtfsFieldDescriptor.Builder fieldBuilder = GtfsFieldDescriptor.builder();
+            fieldBuilder.setName(method.getSimpleName().toString());
+            fieldBuilder.setJavaType(method.getReturnType());
+            FieldType fieldTypeAnnotation = method.getAnnotation(FieldType.class);
+            fieldBuilder.setType(fieldTypeAnnotation != null ? fieldTypeAnnotation.value()
+                    : javaTypeToGtfsType(method.getReturnType()));
+            fieldBuilder.setRequired(method.getAnnotation(Required.class) != null);
+            fieldBuilder.setPrimaryKey(method.getAnnotation(PrimaryKey.class) != null);
+            fieldBuilder.setFirstKey(method.getAnnotation(FirstKey.class) != null);
+            fieldBuilder.setSequenceKey(method.getAnnotation(SequenceKey.class) != null);
+            fieldBuilder.setIndex(method.getAnnotation(Index.class) != null);
+
+            if (method.getAnnotation(Positive.class) != null) {
+                fieldBuilder.setNumberBounds(NumberBounds.POSITIVE);
+            } else if (method.getAnnotation(NonNegative.class) != null) {
+                fieldBuilder.setNumberBounds(NumberBounds.NON_NEGATIVE);
+            } else if (method.getAnnotation(NonZero.class) != null) {
+                fieldBuilder.setNumberBounds(NumberBounds.NON_ZERO);
+            }
+
+            ForeignKey foreignKey = method.getAnnotation(ForeignKey.class);
+            if (foreignKey != null) {
+                fieldBuilder.setForeignKey(ForeignKeyDescriptor.create(
+                        foreignKey.table(), FieldNameConverter.javaFieldName(foreignKey.field())));
+            }
+
+            DefaultValue defaultValue = method.getAnnotation(DefaultValue.class);
+            if (defaultValue != null && defaultValue.value() != null) {
+                fieldBuilder.setDefaultValue(defaultValue.value());
+            }
+
+            fileBuilder.fieldsBuilder().add(fieldBuilder.build());
+        }
+        return fileBuilder.build();
+    }
+
+    FieldTypeEnum javaTypeToGtfsType(TypeMirror javaType) {
+        return javaType.accept(new SimpleTypeVisitor8<FieldTypeEnum, Void>() {
+            @Override
+            public FieldTypeEnum visitPrimitive(PrimitiveType t, Void p) {
+                switch (t.getKind()) {
+                    case BOOLEAN:
+                        return FieldTypeEnum.BOOLEAN;
+                    case BYTE:
+                    case SHORT:
+                    case INT:
+                    case LONG:
+                    case CHAR:
```

Comment:
Is it correct to have `CHAR` here? It seems out of place with the other fields to return as `INTEGER`.